### ISSUE TITLE
API group name need update

### ISSUE THREAD

**Issue by pedjak** [redhat-developer/service-binding-operator#857]
TL;DR: ServiceBinding API group needs to be changed - OLM does not allow operators to define webhooks for API group `operators.coreos.com` and that is by OLM design.

As a part of the work on https://github.com/redhat-developer/service-binding-operator/pull/784 we need to define a mutating web hook, so that we can gather information about requester. Webhooks are also the k8s way to go in cases when we need to validate our ServiceBindings (we have already a few issues related to that)

Unfortunately, I have stumbled upon a blocker, and the issue can be solved only by changing ServiceBinding API group. Currently, it is set to 'operators.coreos.com' and that group is particularly protected by OLM, even OpenShift documentation mentions it:

* https://docs.openshift.com/container-platform/4.6/operators/operator_sdk/osdk-generating-csvs.html#olm-webhook-considerations_osdk-generating-csvs - look for "Admission webhook rules constraints"

* the OLM code ref: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/install/webhook.go#L29

As a result, ServiceBinding CSV gets marked as "Failed" and the operator cannot be installed.

It is a good thing that we still did not reach GA, and our api is still marked as v1alpha1. Group change can certainly be disruptive for some, but that is not the hard part.
